assets/src/edit-story/utils/resourceList.js

```diff
@@ -19,6 +19,18 @@
  * { type: string (cached|smallest|fullsize), url: string }
  */
 
-const resourceList = {};
+const resourceList = {
+  list: {},
+  get: function (id) {
+    return this.list[id];
+  },
+  set: function (id, value) {
+    if (value?.type === 'cached' && this.list[id]) {
+      // We already have better (or equal) resource than cached, prevent flickering
+      return;
+    }
+    this.list[id] = value;
+  },
```

Comment:
Could we unit test this? It'll be super simple, but just to document the desired behavior twice.